### DISCUSSION TITLE
Extract value of an attribute from a chosen attribute

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#1096] [Help] [answered]
I need support to solve the following problem:
I have a node with a parameter called FormulaType in the first attribute of a node, and different formulas in the other attributes of the same node, I want to extract the value of the chosen attribute (value in the first attribute) and read it in the node text.

before asking this forum I tried chatGPT with no result: read the code below.

// Replace 'your_node' with the node you want to extract the value from
def node = your_node

// Get the value of the chosen attribute
def formulaTypeAttribute = node.getAttributeAt(0)      

before of this, I was trying the node['attr'].to.num0 that is working perfectly but if I want to put an **attrId** instead of the name, which is the syntax I need to use? the formula node[attrId].to.num0 doesn't work... do you have any hints?
Thanks
def formulaTypeValue = formulaTypeAttribute.getValue()

// Get the value of the corresponding formula attribute
def formulaAttribute = node.getAttribute(formulaTypeValue)
def formulaValue = formulaAttribute.getValue()

// Update the node text with the formula value
def nodeText = node.getText() + " (" + formulaValue + ")"
controller.getNodeView().setNodeText(node, nodeText)

**Comment by Illegit** [comment]
Sorry I rewrite the statement.

I need support to solve the following problem:
I have a node with a parameter called FormulaType in the first attribute of a node, and different formulas in the other attributes of the same node, I want to extract the value of the chosen attribute (value in the first attribute) and read it in the node text.

before asking this forum I tried chatGPT with no result: read the code below.
-----------------------------------
// Replace 'your_node' with the node you want to extract the value from
def node = your_node

// Get the value of the chosen attribute
def formulaTypeAttribute = node.getAttributeAt(0)
def formulaTypeValue = formulaTypeAttribute.getValue()

// Get the value of the corresponding formula attribute
def formulaAttribute = node.getAttribute(formulaTypeValue)
def formulaValue = formulaAttribute.getValue()

// Update the node text with the formula value
def nodeText = node.getText() + " (" + formulaValue + ")"
controller.getNodeView().setNodeText(node, nodeText)

-------------------------------------
before of this, I was trying the node['attr'].to.num0 that is working perfectly but if I want to put an attrId instead of the name, which is the syntax I need to use? the formula node[attrId].to.num0 doesn't work... do you have any hints?
Thanks

**Comment by macmarrum** [comment]
Could you please attach a mindmap with concrete values and which contains the desired text of your node after the operation?

Based on your description I am guessing the following should do the trick
```groovy
def attributeName = node['FormulaType'].text
node.text = node[attributeName].text
```

**Reply by Illegit** [reply]
I tried to utilize it but doesn't work.
Below my attachment.
Thanks

**Comment by Illegit** [comment]
[TryMap01.zip](https://github.com/freeplane/freeplane/files/11239963/TryMap01.zip)
Thank you macmarrum, here is the map with the node.

**Reply by macmarrum** [reply, marked as answer]
Here's the script.

This one takes the calculated value
```groovy
def attributeNamePrefix = node['FormulaType'].text
def matchingAttribute = node.attributes.find { it.key.startsWith(attributeNamePrefix) }
if (matchingAttribute) {
    node.text = node[matchingAttribute.key].text
}
```

This one takes the formula (raw text)
```groovy
def attributeNamePrefix = node['FormulaType'].text
def matchingAttribute = node.attributes.find { it.key.startsWith(attributeNamePrefix) }
if (matchingAttribute) {
    node.text = matchingAttribute.value
}
```

**Comment by Illegit** [comment]
Hi macmarrum, thank you, 
the first is working well, I have 1 topic: when I click the node I cannot read the script but only the formula, so my question where is the script?
about the second: in this cas I was expecting to read in detail something similar to: 2_MLTxyz, m/s, s. But instead, there is the formula. Do you have any suggestions?

**Comment by Illegit** [comment]
Ok I understood you gave me an alternative to the previous one. Thanks

**Comment by Illegit** [comment]
Macmarrum, I do not understand it was working for some minutes and suddenly remained blocked, because when I change the value from to 2, there isn't any update, but some minutes ag o it was working...
Please try by yourself.

**Reply by macmarrum** [reply]
I'm not sure what you mean.

When I change `node['FormulaType']` to `1` and run the script, I get `9.00` in core text.
When I change `node['FormulaType']` to `2` and run the script, I get `12.00` in core text.

Without a description of your steps, it's difficult for me to help you.

**Reply by Illegit** [reply]
[TryMap02.zip](https://github.com/freeplane/freeplane/files/11242861/TryMap02.zip)
Hi macmarrum, here is my solution using your formula.
Thanks for your help.
